lib/org/chromium/webidl/HTMLFileContents.js

```
@@ -0,0 +1,33 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'HTMLFileContents',
+
+  documentation: 'An HTML file that stores it contents.',
+
+  ids: ['url', 'timestamp'],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'url',
+      required: true,
+      final: true,
```

Here and elsewhere: May not be able to get away with "final: true" anymore; `DatastoreDAO` (which we will use eventually) instantiates objects first, then sets properties. I think "final: true" will create setters that silently fail.